### ISSUE TITLE
Enable ADModule to work on Linux/macOS

### ISSUE THREAD

**Issue by SteveL-MSFT** [PowerShell/PowerShellModuleCoverage#7]
This may be possible after https://github.com/dotnet/corefx/issues/24843 gets completed

Currently, on macOS:

```output
Get-ADUser : System.DirectoryServices.Protocols is not supported on this platform.
```

**Comment by doctordns** [comment]
My understanding is that this module does not work on Linux or Mac for PowerShell 7.

The workaround is to remote into a DC and use implicit remoting.

**Comment by birojnayak** [comment]
@SteveL-MSFT hi Steve, is this doable ? Since S.DS.P (System.DirectoryServices.Protocol) supported in .NET 6  ? If yes, are you willing to take contribution for next version of PS release ? Let us know if there are other challenges...

**Comment by blowdart** [comment]
SDP is "sort of supported". It's ended up using LDAP for compatibility, so the functions that are supported are general lookups for group membership etc. It's not a full blown DS implementation that managing an AD would probably require.

**Comment by birojnayak** [comment]
@blowdart thanks Barry, we can understand.. with existing limitation if there is a list which modules could be ported to PS Core, we would be happy to look/contribute and see if there is a pathway to make it available in newer release..

**Comment by SteveL-MSFT** [comment]
@theJasonHelmick can you follow-up with the AD team?

**Comment by theJasonHelmick** [comment]
@SteveL-MSFT Yes - I've spoken to them before regarding this - I'll give them a call and see if this helps.

**Comment by iSazonov** [comment]
I guess AD module based on Windows wldap32.dll and it is unlikely that the owner will port this while there are already ported libraries.

The .Net Runtime has already been partially ported https://github.com/dotnet/runtime/pull/35380
But there are still compatibility issues https://github.com/dotnet/runtime/issues/36947

I suppose we could make a new module. See the proposal  https://github.com/PowerShell/PowerShell/issues/15578#issuecomment-1240182526
